codec/unknownproto/unknown_fields.go

```diff
@@ -441,3 +441,65 @@ func (d DefaultAnyResolver) Resolve(typeURL string) (proto.Message, error) {
 	}
 	return reflect.New(mt.Elem()).Interface().(proto.Message), nil
 }
+
+// fieldTypeToProtowireType converts a descriptorpb.FieldDescriptorProto_Type to a protowire.Type.
+func fieldTypeToProtowireType(fieldType descriptorpb.FieldDescriptorProto_Type) protowire.Type {
+	switch fieldType {
+	// varint encoded
+	case descriptorpb.FieldDescriptorProto_TYPE_INT64,
+		descriptorpb.FieldDescriptorProto_TYPE_UINT64,
+		descriptorpb.FieldDescriptorProto_TYPE_INT32,
+		descriptorpb.FieldDescriptorProto_TYPE_UINT32,
+		descriptorpb.FieldDescriptorProto_TYPE_BOOL,
+		descriptorpb.FieldDescriptorProto_TYPE_ENUM,
+		descriptorpb.FieldDescriptorProto_TYPE_SINT32,
+		descriptorpb.FieldDescriptorProto_TYPE_SINT64:
+		return protowire.VarintType
+	// fixed64 encoded
+	case descriptorpb.FieldDescriptorProto_TYPE_DOUBLE,
+		descriptorpb.FieldDescriptorProto_TYPE_FIXED64,
+		descriptorpb.FieldDescriptorProto_TYPE_SFIXED64:
+		return protowire.Fixed64Type
+	// fixed32 encoded
+	case descriptorpb.FieldDescriptorProto_TYPE_FLOAT,
+		descriptorpb.FieldDescriptorProto_TYPE_FIXED32,
+		descriptorpb.FieldDescriptorProto_TYPE_SFIXED32:
+		return protowire.Fixed32Type
+	// bytes encoded
+	case descriptorpb.FieldDescriptorProto_TYPE_STRING,
+		descriptorpb.FieldDescriptorProto_TYPE_BYTES,
+		descriptorpb.FieldDescriptorProto_TYPE_MESSAGE,
+		descriptorpb.FieldDescriptorProto_TYPE_GROUP:
+		return protowire.BytesType
+	default:
+		panic(fmt.Sprintf("unknown field type %s", fieldType.String()))
```

Comment:
The %s format specifier always invokes the .String() method, so it is redundant to invoke it manually